### DISCUSSION TITLE
mask passwords

### DISCUSSION THREAD

**Discussion by ariel256** [netbox-community/netbox#12940] [Help Wanted!]
Hello
i am looking for a method to mask passwords that netbox users add to their devices / interfaces.

for example user is adding into the device comment the ssh login credentials (in clear text)
other user created a tag (text) in side the device, and he write there the username and password.

if there is a method to mask a field or comment, or to maybe create a link to a secured vault, or any other more secured method then clear test data?

please advice

**Comment by DanSheps** [comment]
This isn't really something that NetBox can realistically solve.  This is a policy and procedure problem

**Comment by pobradovic08** [comment]
As @DanSheps said, there's no way to prevent users writing passwords in comments or journals, as NetBox does not know the context of that user input and there's no way to detect if that input contains a password or just random text. Correct way would be to store the credentials in a password manager.

**Comment by abhi1693** [comment]
You can look into https://github.com/Onemind-Services-LLC/netbox-secrets however you still won't be able to stop users from not writing the credentials in clear text elsewhere.
